sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBuffer.scala

```diff
@@ -94,6 +112,11 @@ trait RapidsBuffer extends AutoCloseable {
 
   val spillCallback: RapidsBuffer.SpillCallback
 
+  val unspillCallback: RapidsBuffer.UnspillCallback
```

Comment:
Can we just combine this with spillCallback. We can detect from the API if the data is going to the `DEVICE` or not, and it would make adding in metrics to show unspill behavior a lot cleaner.

Reply:
Done.